pulp_ansible/app/galaxy/v3/views.py

```diff
@@ -434,24 +439,55 @@ def get_queryset(self):
         """
         distro_content = self._distro_content
 
-        return CollectionVersion.objects.select_related("content_ptr__contentartifact").filter(
-            pk__in=distro_content
+        return CollectionVersion.objects.filter(pk__in=distro_content).values(
+            "pk",
+            "name",
+            "namespace",
+            "version",
+            "requires_ansible",
+            "authors",
+            "contents",
+            "dependencies",
+            "description",
+            "documentation",
+            "homepage",
+            "issues",
+            "license",
+            "repository",
+            "collection_id",
+            "collection__pulp_created",
+            "collection__pulp_last_updated",
+            "contentartifact__artifact__sha256",
+            "contentartifact__artifact__size",
+        )
+
+    @staticmethod
+    def construct_full_collection_version(obj):
+        """Constructs a new CollectionVersion object."""
+        collection_attrs = {
+            "pk": obj["collection_id"],
+            "pulp_created": obj.pop("collection__pulp_created"),
+            "pulp_last_updated": obj.pop("collection__pulp_last_updated"),
+            "name": obj["name"],
+            "namespace": obj["namespace"],
+        }
+        collection = Collection(**collection_attrs)
+        artifact = Artifact(
+            obj.pop("contentartifact__artifact__sha256"), obj.pop("contentartifact__artifact__size")
         )
+        collection_version = CollectionVersion(**obj)
+        setattr(collection_version, "collection", collection)
+        content_artifact = ContentArtifact(collection_version.relative_path, artifact)
+        setattr(collection_version, "contentartifact", content_artifact)
+        return collection_version
 
     def list(self, request, *args, **kwargs):
         """
         Returns paginated CollectionVersions list.
         """
-        queryset = self.filter_queryset(self.get_queryset())
-        queryset = sorted(
-            queryset, key=lambda obj: semantic_version.Version(obj.version), reverse=True
```

Comment:
we need to sort by version

Reply:
Why though? It's just a json dict that we are returning, does galaxy expect the dictionary to be ordered?

Reply:
I don't know exactly the `ansible-galaxy` CLI code, but I think it expects the versions to be ordered:
https://galaxy.ansible.com/api/v2/collections/pulp/pulp_installer/versions/

Reply:
But this endpoint is only for syncing, correct? They would still use the other endpoints that sort by version for installing collections.

Reply:
Actually I checked the code and they just loop through versions 
https://github.com/ansible/ansible/blob/devel/lib/ansible/galaxy/api.py#L767-L847

Reply:
you are right! `UnpaginatedCollectionVersionViewSet` is only used on sync

Reply:
Dropping the ordering requirement probably would make it go faster.

Reply:
and I agree it's only used on sync